frontend/catalyst/pennylane_extensions.py

```diff
@@ -63,10 +63,10 @@ def __init__(self, fn, device):
 
     def __call__(self, *args, **kwargs):
         if isinstance(self, qml.QNode):
-            if self.device.short_name != "lightning.qubit":
-                raise CompileError(
-                    "Only the lightning.qubit device is supported for compilation at the moment."
-                )
+            # if self.device.short_name != "lightning.qubit":
+            #     raise CompileError(
+            #         "Only the lightning.qubit device is supported for compilation at the moment."
+            #     )
```

Comment:
Just a quick comment, but lets keep this in as _temporary_ solution, but with the allowed devices hardcoded. E.g.,

```python
ALLOWED_DEVICES = ("lightning.qubit", ...)

...

if isinstance(self, qml.QNode):
    if self.device.short_name not in ALLOWED_DEVICES:
        raise CompileError(
            "Only the lightning.qubit device is supported for compilation at the moment."
        )
```

This will ensure that, until we have errors able to be propagated from the runtime, users who (for example) have mistakes/mispellings such as `"ligtning.qubit"` won't get a segfault that is hard to debug.

Reply:
Agreed, mentioned it as well in #86